### PR TITLE
Com 1445 badge icons

### PR DESCRIPTION
**Pull request by GloriaTR**
## Please delete checklist items and sections that are not relevant

**Checklist**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Please check if the PR fulfills these requirements**

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

**Other information**:

**Storybook Link** (Link to a local storybook url to review new functionality)

### CODE REVIEW

File: packages/components/src/components/dataDisplay/Badge/Badge.stories.tsx [hunk 1]
```diff
@@ -58,10 +77,19 @@ const AllTemplate = () => {
 
 export const All = AllTemplate.bind({});
 
-const Template = (args) => <Badge {...args}></Badge>;
+const Template = ({ showLeadingICon, showTrailingIcon, ...args }) => (
+  <Badge
+    {...(showLeadingICon && { leadingIcon: <Icon /> })}
+    {...(showTrailingIcon && { trailingIcon: <Icon /> })}
```
Comment (guilean) [inline packages/components/src/components/dataDisplay/Badge/Badge.stories.tsx:83]: I'm not sure `<Icon />` is actually rendering an icon based on the storybook generated. Could you try using a different icon?

File: packages/components/src/components/dataDisplay/Badge/Badge.stories.tsx [hunk 2]
```diff
@@ -58,10 +77,19 @@ const AllTemplate = () => {
...
+  />
+);
+
 export const Playground = Template.bind({});
 Playground.args = {
   variant: 'solid',
   status: 'primary',
   children: 'Composer badge!',
+  showLeadingIcon: true,
+  showTrailingIcon: true,
```
Comment (guilean) [inline packages/components/src/components/dataDisplay/Badge/Badge.stories.tsx:94]: I've found an issue when trying the playground arguments. I haven't been able to activate both leading and trailing icon in the story. That could be related to the icon issue I mentioned above

File: packages/components/src/components/dataDisplay/Badge/Badge.tsx [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+import { Flex } from '@components';
 import { BadgeProps } from './types';
 
-export const Badge: FC<BadgeProps> = (props) => <ChakraBadge {...props} />;
+export const Badge: FC<BadgeProps> = ({ leadingIcon, trailingIcon, children, ...props }) => {
+  const leftIcon = getIcon(leadingIcon);
+  const rightIcon = getIcon(trailingIcon);
+  return (
+    <ChakraBadge {...props}>
+      <Flex alignItems="center" gap="spacer-1">
```
Comment (guilean) [inline packages/components/src/components/dataDisplay/Badge/Badge.tsx:11]: Could you try moving these styles to the `ChakraBadge`?

**Review by guilean — CHANGES_REQUESTED** [review]

File: packages/components/src/components/dataDisplay/Badge/Badge.stories.tsx (hunk 1 above)
Comment (GloriaTR) [reply packages/components/src/components/dataDisplay/Badge/Badge.stories.tsx]: Fixed!

File: packages/components/src/components/dataDisplay/Badge/Badge.stories.tsx (hunk 2 above)
Comment (GloriaTR) [reply packages/components/src/components/dataDisplay/Badge/Badge.stories.tsx:102]: I had a typo in showLeadingIcon in the Template function. Now this is fixed. I also set the option to enable both icons

File: packages/components/src/components/dataDisplay/Badge/Badge.tsx (hunk 3 above)
Comment (GloriaTR) [reply packages/components/src/components/dataDisplay/Badge/Badge.tsx]: Fixed!

Comment (guilean) [reply packages/components/src/components/dataDisplay/Badge/Badge.tsx]: Is the `inline-flex` needed? Or would it work with display flex?

**Review by guilean — APPROVED** [review]

**Pull request merged**